metaworld/envs/mujoco/sawyer_xyz/sawyer_peg_insertion_side_v2.py

```diff
@@ -104,24 +104,17 @@ def _set_obj_xyz(self, pos):
 
     def reset_model(self):
         self._reset_hand()
-        self.sim.model.body_pos[self.model.body_name2id('box')] = self.goal.copy()
+
+        self.sim.model.body_pos[self.model.body_name2id('box')] = np.array([-0.3, 0.6, 0.05])
         self._state_goal = self.sim.model.site_pos[self.model.site_name2id('hole')] + self.sim.model.body_pos[self.model.body_name2id('box')]
         self.obj_init_pos = self.init_config['obj_init_pos']
         self.objHeight = self.get_body_com('peg').copy()[2]
         self.heightTarget = self.objHeight + self.liftThresh
 
         if self.random_init:
-            goal_pos = np.random.uniform(
-                self.obj_and_goal_space.low,
-                self.obj_and_goal_space.high,
-                size=(self.obj_and_goal_space.low.size),
-            )
+            goal_pos = self._get_state_rand_vec()
             while np.linalg.norm(goal_pos[:2] - goal_pos[-3:-1]) < 0.1:
-                goal_pos = np.random.uniform(
-                    self.obj_and_goal_space.low,
-                    self.obj_and_goal_space.high,
-                    size=(self.obj_and_goal_space.low.size),
-                )
+                goal_pos = self._get_state_rand_vec()
```

Comment:
doesn't it make more sense to put the rejection sampling (the while loop) inside `_get_state_and_rand_vec()`?

Reply:
I just copied how Avnish did it in the other environments. `_get_state_rand_vec()` is in the base class, but rejection handling is unique to the environment. We could override it in the future, but for now I just wanted to match the v1 envs.